consensus/blake3/consensus.go

```diff
@@ -703,6 +703,60 @@ func (blake3 *Blake3) GetLinkExternalBlocks(chain consensus.ChainHeaderReader, h
 	return externalBlocks, nil
 }
 
+// PreviousCoincidentOnPath searches the path for a block of specified order in the specified slice
+//     *slice - The zone location which defines the slice in which we are validating
+//     *order - The order of the conincidence that is desired
+//     *path - Search among ancestors of this path in the specified slice
+func (blake3 *Blake3) PreviousCoincidentOnPath(chain consensus.ChainHeaderReader, header *types.Header, slice []byte, order, path int) (common.Hash, error) {
+
+	if err := chain.CheckContextAndOrderRange(path); err != nil {
+		return common.Hash{}, err
+	}
+	if err := chain.CheckContextAndOrderRange(order); err != nil {
+		return common.Hash{}, err
+	}
+	if err := chain.CheckLocationRange(slice); err != nil {
+		return common.Hash{}, err
+	}
+
+	for {
+		// If block header is Genesis return it as coincident
+		if header.Number[path].Cmp(big.NewInt(0)) <= 0 {
```

Comment:
We should also check order here. `Number <= 0` obviously stops the search (can't search past 0), but we should still check the block matches all the conditions.

Perhaps instead, it would be better to just put this check at the end of the loop e.g. @ L756:
```
// If we hit block 0 and still didn't find a coincident, then this chain has an invalid genesis block
if header.Number[path].Cmp(big.NewInt(0)) <= 0 {
    // Stop search and return error
}
```

The reason to do it this way, is to avoid an edge case, where a block claims to be block 0, and we just blindly assume it is genesis and short circuit. That error wouldn't make it past PCRC (as implemented today), but it'd be better if that case didn't make it through this function in the first place, IMO.

Reply:
Will change the comparison on Number to ==

We need to keep this at the top of the loop else the subsequent checks of finding parents will fail.  

We can't easily ensure the issue with order.  However, we can make the genesis shared between all chains as referenced in this issue: https://github.com/spruce-solutions/go-quai/issues/290

This will prevent PCRC from failing at start without having to make a bunch of special cases.